src/repository/cartRepository.ts

```diff
@@ -0,0 +1,33 @@
+import { Cart, ICart } from '../models/Cart';
+import { Types } from 'mongoose';
+
+export class CartRepository {
+  // Отримати корзину користувача за ID
+  async getCartByUserId(userId: string): Promise<ICart | null> {
+    return Cart.findOne({ userId })
+      .populate('products')  // Поповнення продуктів
+      .populate('userId')    // Поповнення інформації про користувача
+      .exec(); // Поповнення продуктів
+  }
+
+  // Додати продукт до корзини
+  async addToCart(userId: string, productId: string): Promise<ICart | null> {
+    const cart = await this.getCartByUserId(userId);
+
+    // Перетворення productId на ObjectId
+    const productObjectId = new Types.ObjectId(productId);
+
+    if (!cart) {
+      const newCart = new Cart({ userId, products: [productObjectId] });
+      return newCart.save();
+    }
+
+    cart.products.push(productObjectId);
+    return cart.save();
```

Comment:
Also it is possible to use `$push` operator with `findByIdAndUpdate()`
[findByIdAndUpdate](https://www.geeksforgeeks.org/how-to-push-item-from-an-array-in-mongoose/)